app/src/sharedTest/java/org/oppia/app/profile/AdminAuthActivityTest.kt

```diff
@@ -132,6 +138,80 @@ class AdminAuthActivityTest {
     }
   }
 
+  @Test
+  fun testAdminAuthActivity_buttonState_configurationChanged_buttonStateIsPreserved() {
+    launch<AdminAuthActivity>(
+      AdminAuthActivity.createAdminAuthActivityIntent(
+        context,
+        "12345",
+        -10710042
+      )
+    ).use {
+      onView(allOf(withId(R.id.input), isDescendantOfA(withId(R.id.input_pin)))).perform(
+        typeText("12345"),
+        closeSoftKeyboard()
+      )
+      onView(isRoot()).perform(orientationLandscape())
+      onView(withId(R.id.submit_button)).check(
```

Comment:
This test case is failing because of drawable checking, so we have another way to test is functionality by checking whether the button is enabled or not.

`onView(withId(R.id.submit_button)).check(matches(isEnabled()))`

So your final code should be 
```
  @Test
  fun testAdminAuthActivity_buttonState_configurationChanged_buttonStateIsPreserved() {
    launch<AdminAuthActivity>(
      AdminAuthActivity.createAdminAuthActivityIntent(
        context,
        "12345",
        -10710042
      )
    ).use {
      onView(allOf(withId(R.id.input), isDescendantOfA(withId(R.id.input_pin)))).perform(
        typeText("12345"),
        closeSoftKeyboard()
      )
      onView(isRoot()).perform(orientationLandscape())
      onView(withId(R.id.submit_button)).check(matches(isEnabled()))
    }
```

Reply:
@rt4914 I have updated the test case. But could you please explain what went wrong with the drawable matching part because of which the test was failing.

Reply:
Actually that drawable matching code was a custom code which mainly works in images only and because of which it was failing.